### ISSUE TITLE
Release 4.9.1 - RC 1 - E2E UX tests - File Integrity Monitoring

### ISSUE THREAD

**Issue by juliamagan** [wazuh/wazuh#25827]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.9.1-rc1/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-agent-div2 team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Sep 23, 2024 and notify the @wazuh/devel-agent-div2 team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-agent-div2 team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Sep 24, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Sep 25, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Quickstart](https://documentation-dev.wazuh.com/v4.9.1-rc1/quickstart.html) | - | RHEL 7 x86_64 |
| Server | Same as indexer, all-in-one | - | - |
| Dashboard | Same as indexer, all-in-one | - | - |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.9.1-rc1/installation-guide/wazuh-agent/index.html) | - | Windows 10 x86_64, Debian 12 aarch64, RHEL 9 x86_64, macOS Ventura arm |

## Test description

Test different FIM use cases for Windows, Linux and macOS:
- whodata/realtime (scheduled if the others are not supported)
- report_changes
- inventory of files

Check that this uses cases still work for the current release under test:
https://documentation-dev.wazuh.com/v4.9.1-rc1/proof-of-concept-guide/poc-file-integrity-monitoring.html

Check that this blog post is still valid for the current release under test (suggest changes otherwise):
https://wazuh.com/blog/preventing-and-detecting-ransomware-with-wazuh/

Navigate trough WUI - FIM section to ensure that data is accurate and updated when needed (inventory/alerts/dashboards)

## Known issues

- https://github.com/wazuh/wazuh/issues/8602
- https://github.com/wazuh/wazuh-documentation/issues/5013 
- https://github.com/wazuh/wazuh/issues/18429
- https://github.com/wazuh/wazuh/issues/16226
- https://github.com/wazuh/wazuh/issues/10211
- https://github.com/wazuh/wazuh/issues/21130

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below. **REMOVE CURRENT EXAMPLES**:

#### Deployment

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢 | Deployment environment | - | - |
| 🟢 | Agent installation - Windows | -  | - |
|🟢 |Agent installation - Debian 12 aarch64 |- | - |
|🟢 |Agent installation - RHEL 9 x86_64|- | - |
|🟢 |Agent installation - macOS Ventura arm |- |-|

#### FIM use cases and POC

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
|🟢 |Windows - Whodata|- | - | 
|🟢 |Windows - Report changes |- | - | 
|🟢 |Windows - File inventory|- | - |
|🟢 |Windows - Dashboard|- | - |
|🟡 |macOS Ventura arm - Whodata| Known Issue, whodata only works on Linux and Windows | #8602 |
|🟢 |macOS Ventura arm - Report changes|- | - |
|🟢 |macOS Ventura arm - File Inventory|- | - |
|🟢 |macOS Ventura arm - Dashboard|- | - |
|🟢 |Debian 12 aarch64 - Whodata|- | - | 
|🟢 |Debian 12 aarch64 - Report changes |- | - | 
|🟢 |Debian 12 aarch64 - File inventory|- | - |
|🟢 |Debian 12 aarch64 - Dashboard|- | - |
|🟢 |RHEL 9 x86_64 - Whodata|- | - | 
|🟢 |RHEL 9 x86_64 - Report changes |- | - | 
|🟢 |RHEL 9 x86_64 - File inventory|- | - |
|🟢 |RHEL 9 x86_64 - Dashboard|- | - |
|🟢 |Windows - POC|- | - |
|🟢 |Linux - POC|- | - |

#### Preventing and detecting ransomware with Wazuh

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
|🟢 |Step 1: Prepare the test environment|- | - |
|🟢 |Step 2: Simulating the attack|- | - |
|🟡 |Monitoring Wazuh alerts and setting up triggers|The blogpost was published with an older version of Wazuh that use OpenDistro, not Wazuh Dashboard | https://github.com/wazuh/wazuh-documentation/issues/5013 |

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - None
- Did you face any challenges not covered by the guideline?
  - None
- Suggestions for improvement:
  - No suggestion

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @juliamagan
- [x] @wazuh/devel-agent-div2

**Comment by ace-109** [comment]
# Deployment environment

The central components were deployed on a `RHEL 7 x86_64` server using quick start.

## Quickstart installation on RHEL 7
```
[root@localhost wazuh]# cat /etc/*release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.9 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.9"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.9 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.9:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.9"
Red Hat Enterprise Linux Server release 7.9 (Maipo)
Red Hat Enterprise Linux Server release 7.9 (Maipo)
```
```
curl -sO https://packages-dev.wazuh.com/4.9/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
```
The concluding part of the installation script shows it successfully installed
```
23/09/2024 15:11:03 INFO: Starting service filebeat.
23/09/2024 15:11:03 INFO: filebeat service started.
23/09/2024 15:11:03 INFO: --- Wazuh dashboard ---
23/09/2024 15:11:03 INFO: Starting Wazuh dashboard installation.

23/09/2024 15:14:37 INFO: Wazuh dashboard installation finished.
23/09/2024 15:14:37 INFO: Wazuh dashboard post-install configuration finished.
23/09/2024 15:14:37 INFO: Starting service wazuh-dashboard.
23/09/2024 15:14:38 INFO: wazuh-dashboard service started.
23/09/2024 15:14:38 INFO: Updating the internal users.
23/09/2024 15:14:45 INFO: A backup of the internal users has been saved in the /etc/wazuh-indexer/internalusers-backup folder.
23/09/2024 15:14:59 INFO: The filebeat.yml file has been updated to use the Filebeat Keystore username and password.
23/09/2024 15:15:44 INFO: Initializing Wazuh dashboard web application.
23/09/2024 15:15:44 INFO: Wazuh dashboard web application initialized.
23/09/2024 15:15:44 INFO: --- Summary ---
23/09/2024 15:15:44 INFO: You can access the web interface https://<wazuh-dashboard-ip>:443
    User: admin
    Password: XXXXXX
23/09/2024 15:15:44 INFO: --- Dependencies ---
23/09/2024 15:15:44 INFO: Removing lsof.
23/09/2024 15:15:46 INFO: Installation finished.
```

Login was also successful
![image](https://github.com/user-attachments/assets/80e181b4-cd0e-487b-bf5c-b397cc273a14)

**Comment by ace-109** [comment]
# Agent installation
## Windows
I downloaded the installer package 
```
Invoke-WebRequest -Uri "https://packages-dev.wazuh.com/pre-release/windows/wazuh-agent-4.9.1-1.msi" -OutFile "wazuh-rc1.msi"
```
```
.\wazuh-rc1.msi /q WAZUH_MANAGER=1.1.X.X
```
I started the service
````
NET START Wazuh
```

Agent is successfully enrolled
![image](https://github.com/user-attachments/assets/555ad4d5-824f-4fcb-9029-d2b445ce9d3a)
There are some events from the endpoint
![image](https://github.com/user-attachments/assets/c59d1152-e91a-4d63-b810-351875b9c1b1)

**Comment by ace-109** [comment]
# FIM use cases

I tested the following use cases on the endpoints.
- whodata
- realtime / scheduled ( when realtime or whodata is not supported)
- report_changes
- inventory of files

**Comment by ace-109** [comment]
# Preventing and detecting ransomware with Wazuh

I carried out this test using the Debian aarch64 machine.
I created the `/home/vagrant/test` directory,
```
mkdir -p /home/vagrant/test
```
The `ossec.conf` file was updated to monitor the directory
```
<syscheck>
    <directories check_all="yes" whodata="yes">/home/vagrant/test</directories>
</syscheck>
```
I restarted the Wazuh agent
```
systemctl restart wazuh-agent
```
I downloaded the python script
```
root@ip-172-31-41-50:/home/admin# root@ip-172-31-41-50:/home/admin# wget https://wazuh.com/resources/blog/detect-ransomware-with-wazuh/wazuh-ransomware-poc.py
--2024-09-24 10:56:46--  https://wazuh.com/resources/blog/detect-ransomware-with-wazuh/wazuh-ransomware-poc.py
Resolving wazuh.com (wazuh.com)... 52.85.132.32, 52.85.132.4, 52.85.132.84, ...
Connecting to wazuh.com (wazuh.com)|52.85.132.32|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3559 (3.5K) [binary/octet-stream]
Saving to: ‘wazuh-ransomware-poc.py’

wazuh-ransomware-poc.py      100%[==============================================>]   3.48K  --.-KB/s    in 0s

2024-09-24 10:56:46 (55.3 MB/s) - ‘wazuh-ransomware-poc.py’ saved [3559/3559]

root@ip-172-31-41-50:/home/admin# ls
wazuh-ransomware-poc.py
```

**Comment by jotacarma90** [comment]
LGTM

**Comment by rauldpm** [comment]
# Final reviews

> The concluding part of the installation script shows it successfully installed

@ace-109 please do not skip deployment output, we should report all outputs just in case something is found or it is not expected and you may not consider it as something wrong

> The UI in the blog is not the same as the new Wazuh dashboard but I could setup the monitors and triggers

If this blog https://wazuh.com/blog/preventing-and-detecting-ransomware-with-wazuh/ does not have an updated screenshot, we should suggest to change it, as it may confuse users, please open an issue

**Comment by ace-109** [comment]
Hello @rauldpm 
The blogpost screenshot is a known issue https://github.com/wazuh/wazuh-documentation/issues/5013, I updated the comments to reference the previously created issue regarding it. The blog was published with V3.1.0 and it had some OpenDistro references.
